zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/X509AuthenticationProvider.java

```diff
@@ -212,11 +213,14 @@ protected String getClientId(X509Certificate clientCert) {
     private String matchAndExtractSAN(X509Certificate clientCert)
         throws CertificateParsingException {
         Integer matchType =
-            Integer.getInteger(ZOOKEEPER_X509AUTHENTICATIONPROVIDER_CLIENTCERTIDSANMATCHTYPE);
+            Integer.getInteger(ZOOKEEPER_X509AUTHENTICATIONPROVIDER_CLIENT_CERT_ID_SAN_MATCH_TYPE);
         String matchRegex =
-            System.getProperty(ZOOKEEPER_X509AUTHENTICATIONPROVIDER_CLIENTCERTIDSANMATCHREGEX);
+            System.getProperty(ZOOKEEPER_X509AUTHENTICATIONPROVIDER_CLIENT_CERT_ID_SAN_MATCH_REGEX);
         String extractRegex =
-            System.getProperty(ZOOKEEPER_X509AUTHENTICATIONPROVIDER_CLIENTCERTIDSANEXTRACTREGEX);
+            System.getProperty(
+                ZOOKEEPER_X509AUTHENTICATIONPROVIDER_CLIENT_CERT_ID_SAN_EXTRACT_REGEX);
+        Integer extractMatcherGroupIndex = Integer.getInteger(
+            ZOOKEEPER_X509AUTHENTICATIONPROVIDER_CLIENT_CERT_ID_SAN_EXTRACT_MATCHER_GROUP_INDEX);
```

Comment:
I think an additional config to control the matching group index is overkill. A simpler and more straightforward way would be to enforce the regex to have exactly one matching group and we always extract matching group # 1. We can always modify the regex to control which part of the input string should be capture group # 1 (so in a way, we can modify the regex to get the same behavior as modifying the index). Please let me know if that reasoning is incorrect.

Reply:
You are not incorrect, but there's value in making this part configurable too - it might make it easier for operators to come up with the right regex and give more flexibility. Since the original motivation for this addendum commit was to make this more flexible, I think this change is appropriate. 

I can make the default index 1.